### PR TITLE
fix: React plugin live query improvements, and export types

### PR DESCRIPTION
**Pull request by samwillis**

### CODE REVIEW

**Comment by samwillis** [comment]
@msfstef this works, I'm using it in the new Linearline PR, but its failing test. Any idea why? I'm can't work it out...

**Review by msfstef — COMMENTED** [review]
Might be good to add a test for the case of parameter argument mutated in place - it's one of those annoying React cases

File: packages/pglite-react/src/hooks.ts [hunk 1]
```diff
@@ -1,15 +1,34 @@
...
   const db = usePGlite()
   const [results, setResults] = useState<Results<T>>()
+
+  // We manually check for changes to params so that we can support as change to the
+  // number of params
+  const paramsRef = useRef<unknown[] | undefined | null>(params)
+  if (!arrayEqual(paramsRef.current as unknown[], params as unknown[])) {
+    paramsRef.current = params
+  }
+  const currentParams = paramsRef.current
```
[the remark this answers is not in the document]
Comment (samwillis) [reply packages/pglite-react/src/hooks.ts:30]: Hmm, do you know a better pattern? I don't want to use JSON.stringify.

the issue was that in Linearlite the filters in the UI build the query that's passed to the file query, and depending on what you set the number of parameters changes. That was throwing an error as useEffect doesn't like the number of dependencies to change.

[the remark this answers is not in the document]
Comment (samwillis) [reply packages/pglite-react/src/hooks.ts:30]: This was the code in Linearlite that cased me to change this - I can see it being a common pattern to build a live query from filters:

https://github.com/electric-sql/pglite/blob/e883ecf130635f756040383ed057442348eacd89/demos/linearlite/src/pages/List/index.tsx#L13-L39

[the remark this answers is not in the document]
Comment (msfstef) [reply packages/pglite-react/src/hooks.ts:30]: hmm imo even in the linearlite snippet the work could probably be done within a `useMemo`.

as for the code for `useLiveQuery`, I think we can have a `useEffect` with the params array as the dependency, that does an equality check and updates a `useState` var `currentParams` - that way we stay within the react component lifecycle. I can push some code and a test for this case.

File: packages/pglite-react/src/hooks.ts [hunk 2]
```diff
@@ -3,13 +3,37 @@ import { Results } from '@electric-sql/pglite'
...
   const [results, setResults] = useState<Results<T>>()
+  const [currentParams, setCurrentParams] = useState(params)
+
+  useEffect(
+    () =>
+      setCurrentParams((currentParams) =>
+        !paramsEqual(currentParams, params) ? params : currentParams,
+      ),
+    [params],
+  )
```
Comment (samwillis) [inline packages/pglite-react/src/hooks.ts:35]: @msfstef I think this useEffect will run after the component has rendered. So we get:

1. Update params
2. Render triggered by change, but it uses value of `currentParams` which is old
3. Effect runs after render
4. Render with new params

**Pull request merged**